File: client/sdk/python/openai/embeddings.py
"""OpenAI /v1/embeddings endpoint functions."""

from openai import OpenAI


def create_client(api_key, base_url):
    return OpenAI(api_key=api_key, base_url=base_url + "/v1")


def create_embedding(text, model, api_key, base_url):
    client = create_client(api_key=api_key, base_url=base_url)
    response = client.embeddings.create(
        model=model,
        input=text,
    )
    embedding = response.data[0].embedding
    usage = getattr(response, "usage", None)
    return {
        "object": "embedding",
        "model": model,
        "dimensions": len(embedding),
        "embedding": embedding,
        "prompt_tokens": getattr(usage, "prompt_tokens", None),
        "total_tokens": getattr(usage, "total_tokens", None),
    }

File: client/sdk/typescript/src/openai/embeddings.ts
import type { ClientConfig } from "../config";

import { createClient } from "./shared";

export type EmbeddingResult = {
  object: string;
  model: string;
  dimensions: number;
  embedding: number[];
  promptTokens: number | null;
  totalTokens: number | null;
};

export async function createEmbedding(
  config: ClientConfig,
  text: string,
  model: string,
): Promise<EmbeddingResult> {
  const client = createClient(config);
  const response = await client.embeddings.create({
    model,
    input: text,
  });
  const embedding = response.data[0]?.embedding ?? [];

  return {
    object: "embedding",
    model,
    dimensions: embedding.length,
    embedding,
    promptTokens: response.usage?.prompt_tokens ?? null,
    totalTokens: response.usage?.total_tokens ?? null,
  };
}
